connection.js

```diff
@@ -54,13 +43,35 @@ function _getConnection(connString = CONN_STRING){
 		});
 }
 
-var connectionsDict = {
-	main: _getConnection(),
-};
-module.exports = function(connString){
-	if (connString && !connectionsDict[connString]) {
-		connectionsDict[connString] = _getConnection(connString);
+function _closeConnection(conn) {
+	var closed = false;
+
+	process.once('SIGINT', () => {
+		if (closed) {
+			_log('close already ran');
+			return;
+		}
+		_log('running close');
+		closed = true;
+		conn.close();
+	});
+
+	return conn;
+}
+
+function _createConnection(fn, connString) {
+	const connName = connString || 'main';
+
+	if (!connectionsDict[connName]) {
+		connectionsDict[connName] = Connection(connString).then(fn);
 	}
 
-	return connString ? connectionsDict[connString] : connectionsDict.main;
-};
+	return connectionsDict[connName];
+}
+
+var connectionsDict = {};
+
+module.exports = {
```

Comment:
if you change your export to look like this (untested) you can ditch all the changes where you `require('...').getConnection`
```javascript
const getConnection = _.partial(_createConnection, _closeConnection);
getConnection.Connection = _.partial(_createConnection, conn =>conn);
module.exports = getConnection;
```